lms/migrations/versions/b99c4d910801_disable_provisioning_for_non-whitelisted_.py

```diff
@@ -0,0 +1,96 @@
+"""
+Disabled provisioning for non-whitelisted consumer keys.
+
+Revision ID: b99c4d910801
+Revises: 7e4124035651
+Create Date: 2018-12-07 12:04:18.524467
+
+"""
+import logging
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+
+log = logging.getLogger(__name__)
+
+
+# revision identifiers, used by Alembic.
+revision = "b99c4d910801"
+down_revision = "7e4124035651"
+
+
+Base = declarative_base()
+Session = sessionmaker()
+
+
+class ApplicationInstance(Base):
+    __tablename__ = "application_instances"
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    consumer_key = sa.Column(sa.String)
+    provisioning = sa.Column(sa.Boolean())
+
+
+def upgrade():
+    session = Session(bind=op.get_bind())
+
+    disabled = 0
+    enabled = 0
+
+    for ai in session.query(ApplicationInstance):
```

Comment:
How does this migration behave if any of the listed keys aren't actually in the DB?

Reply:
It disables the provisioning feature for all non-whitelisted instances. So if none of the the whitelisted instances are in your DB it'll disable provisioning for all your instances.